### DISCUSSION TITLE
How to use DefaultDataFetcherExceptionHandler?

### DISCUSSION THREAD

**Discussion by davekats** [Netflix/dgs-framework#824] [Q&A] [answered]
The documentation [here](https://netflix.github.io/dgs/error-handling/#mapping-custom-exceptions) provides a code snippet that uses the DefaultDataFetcherExceptionHandler class, however it doesn't show what you need to import in order to use the class. I can't find the type anywhere in the library.

**Comment by Cuica20** [comment, marked as answer]
`package com.dsrp.backend.error;

import com.dsrp.backend.exception.DgsNotFoundException;
import com.netflix.graphql.dgs.exceptions.DefaultDataFetcherExceptionHandler;
import com.netflix.graphql.types.errors.ErrorDetail;
import com.netflix.graphql.types.errors.ErrorType;
import com.netflix.graphql.types.errors.TypedGraphQLError;
import graphql.GraphQLError;
import graphql.execution.DataFetcherExceptionHandler;
import graphql.execution.DataFetcherExceptionHandlerParameters;
import graphql.execution.DataFetcherExceptionHandlerResult;
import org.springframework.stereotype.Component;

import java.util.HashMap;
import java.util.Map;

@Component
public class CustomDataFetchingExceptionHandler implements DataFetcherExceptionHandler {
    private final DefaultDataFetcherExceptionHandler defaultHandler = new DefaultDataFetcherExceptionHandler();

    @Override
    public DataFetcherExceptionHandlerResult onException(DataFetcherExceptionHandlerParameters handlerParameters) {
        if (handlerParameters.getException() instanceof DgsNotFoundException) {
            Map<String, Object> debugInfo = new HashMap<>();
            debugInfo.put("Value", handlerParameters.getException().getMessage());

            GraphQLError graphqlError = TypedGraphQLError.newNotFoundBuilder()
                    .message(String.format("No se logro recuperar el dato con el valor %s", handlerParameters.getException().getMessage()))
                    .debugInfo(debugInfo)
                    .errorType(ErrorType.NOT_FOUND)
                    .path(handlerParameters.getPath()).build();
            return DataFetcherExceptionHandlerResult.newResult()
                    .error(graphqlError)
                    .build();
        } else {
            return defaultHandler.onException(handlerParameters);
        }
    }
}`

**Reply by Cuica20** [reply]
`package com.dsrp.backend.exception;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(value = HttpStatus.NOT_FOUND, reason = "data: not found")
@NoArgsConstructor
public class DgsNotFoundException extends RuntimeException {

    @Getter
    @Setter
    private String message = "data: not found";

    public DgsNotFoundException(String message) {
        super();
        this.message = message;
    }

}`

**Reply by dirk-ecker** [reply]
It would be nice to have some text that explains the solution. IMHO the solution is not more than the example that can be found through the provided link. 

The real question is how does DSG know that it should use this new Handler. That has to be configured somewhere or ...

**Reply by dirk-ecker** [reply]
Found out by myself/college. The _component_ annotation does it.
